install-and-configure/install/multi-cluster/federated-etl/thanos-migration-guide.md

```diff
@@ -0,0 +1,155 @@
+* Having enough storage is important and will vary based on environment.
+
+## Prerequisites
+
+To migrate from Thanos multi-cluster federated architecture to Aggregator, users *must* complete the following steps:
+
+### Step 1: Use the existing Thanos object store or create a new dedicated object store
+
+This object store is where the ETL backups will be pushed to from the primary cluster.
+
+### Step 2: Enable [ETL Backups](https://docs.kubecost.com/install-and-configure/install/etl-backup#google-cloud-storage) on the *primary cluster only*. 
+
+This will ensure we have all historical data in durable storage and it will create the directory the ETL Utils container needs in order to create the directory structure needed for Aggregator to pull the ETL from each respective cluster.
+
+```
+kubecostModel:
+  etlBucketConfigSecret: <YOUR_SECRET_NAME>
+```
+
+### Step 3: Validate that an `/etl` directory is present in the object store
+
+There should be ETL data present in the following directories. CloudCosts will only have ETL data if cloud integration is enabled.
+
+* `/etl/bingen/allocations`
+* `/etl/bingen/assets/`
+* `/cloudcosts/`
+
+
+### Step 4: Create a new secret for the federated store
+
+This will point to the existing Thanos object store or the new one created in Step 1.
+
+{% hint style="warning" %}
+The name of the .yaml file used to create the secret *must* be named _federated-store.yaml_ or Aggregator will not start.
+{% endhint %}
+
+
+```
+kubectl create secret generic federated-store --from-file=federated-store.yaml -n kubecost
+```
+
+### Step 5: Set Aggregator and ETL Utils in *values.yaml* on the primary cluster.
+
+{% hint style="warning" %}
+*Do not* disable Thanos during this step. This will not negatively impact the source data. Thanos will be disabled in a later step.
+{% endhint %}
+
+* [Enable Aggregator for the primary cluster](https://docs.kubecost.com/install-and-configure/install/multi-cluster/federated-etl/aggregator). 
+
+Note:  
+
+* ETL Utils
+```
+etlUtils:
+  thanosSourceBucketSecret: kubecost-thanos
+  affinity: {}
+  enabled: true
+  env:
+    LOG_LEVEL: debug
+  fullImageName: gcr.io/kubecost1/cost-model-etl-utils:0.2
+  nodeSelector:
+    kubernetes.io/arch: amd64
+  resources: {}
+  tolerations: {}
+kubecostModel:
+  federatedStorageConfigSecret: kubecost-thanos
+```
+
+### Step 6: Apply the changes and wait for data to populate in the UI. 
+
+Ensure all pods and containers are running:
+
+ ```
+kubecost-cost-analyzer-685fd8f677-k652h        4/4     Running   2 (12m ago)   3h2m
+kubecost-forecasting-6cf4dd7b98-7z8f5          1/1     Running   0             66m
+kubecost-etl-utils-6cdd489596-5dl75           1/1     Running   0          6d20h
+```
+
+This can take some time depending on how many unique containers are running in the environment and how much ETL data is in the object store. A couple importants steps are happening in the background which take time to complete.
```

Comment:
I need to confirm this, but I'm pretty sure that Aggregator has to be running in a separate Pod in order for it to target "uncombined" data in the object store. When running in the single-pod config (within kubecost-cost-analyzer) I think it only read from the local directory. The end-state we want users to be in here is an "enterprise" setup with separate Aggregator StatefulSet.

Reply:
@michaelmdresser gotcha - I thought it would be one of the 4 containers in the cost-analyzer pod. Not sure if thats been finalized.

Reply:
That's the default configuration, but when using an object store for metric federation the config switches to having Aggregator running separately from cost-analyzer, sort of like this:
```
→ k get pod -n kubecost-cloudcost
NAME                                                    READY   STATUS    RESTARTS     AGE
kubecost-cloudcost-aggregator-0                         1/1     Running   0            16h
kubecost-cloudcost-cloud-cost-79b89f66-v5jq2            1/1     Running   0            3d22h
kubecost-cloudcost-cost-analyzer-7f9b79d74-rmrrt        2/2     Running   0            3d16h
kubecost-cloudcost-diagnostics-59f98b8dc9-mv5gz         1/1     Running   0            15h
kubecost-cloudcost-forecasting-6584f99d8d-np46p         1/1     Running   0            15h
```

If that's unclear let's dig more, I want to make sure folks are on the same page about this.